zlib.h

```diff
@@ -109,6 +109,9 @@ typedef struct z_stream_s {
     int                   data_type;  /* best guess about the data type: binary or text
                                          for deflate, or the decoding state for inflate */
     uint32_t              adler;      /* Adler-32 or CRC-32 value of the uncompressed data */
+#if defined(ZLIB_COMPAT) && defined(X86_64)
+    uint32_t              padding;    /* pad out to the same size as the zlib struct */
+#endif
```

Comment:
This is wrong, because member "adler" is not aligned correctly due to "data_type" being first 32-bit member after 64-bit "void" member. When "adler" was "unsigned long/uint64_t", it had extra 4-byte implicit padding added before it. In zlib-ng that implicit padding is unnecessary due to "int" and "uint32_t" both being 4 bytes.